File: crates/flowrt-codegen/tests/golden/feedback_loop_cpp/expected/app/stubs/cpp/plant.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Plant final : public flowrt_app::PlantInterface {
public:
    flowrt::Status on_tick(
        const flowrt::Latest<flowrt_app::Cmd>& cmd,
        flowrt::Output<flowrt_app::State>& state) override {
        (void)cmd;
        state.write(flowrt_app::State{});
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Plant>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/feedback_loop_rust/expected/app/stubs/rust/plant.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Plant;

impl flowrt_app::components::Plant for Plant {
    fn on_tick(
        &mut self,
        cmd: flowrt::Latest<'_, flowrt_app::messages::Cmd>,
        state: &mut flowrt::Output<flowrt_app::messages::State>,
    ) -> flowrt::Status {
        let _ = cmd;
        state.write(flowrt_app::messages::State::default());
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Plant::default()))
}
